core/src/main/java/org/apache/calcite/prepare/PlannerImpl.java

```diff
@@ -297,8 +300,10 @@ public RelRoot expandView(RelDataType rowType, String queryString,
     final SqlConformance conformance = conformance();
     final CalciteCatalogReader catalogReader =
         createCatalogReader().withSchemaPath(schemaPath);
+    final SqlOperatorTable opTab =
+        ChainedSqlOperatorTable.of(operatorTable, catalogReader);
```

Comment:
Can we make more effort to define a method like `createSqlValidator` that used both by `validate` and `expandView` ?

Reply:
Sure, this can help remove duplicate codes. I have added createSqlValidator in a new commit.